bolt/lib/Passes/ADRRelaxationPass.cpp

```diff
@@ -56,9 +56,13 @@ void ADRRelaxationPass::runOnFunction(BinaryFunction &BF) {
           continue;
       }
 
-      BinaryFunction *TargetBF = BC.getFunctionForSymbol(Symbol);
-      if (TargetBF && TargetBF == &BF)
-        continue;
+      // Don't relax adr if it points to the same function and it is not split
+      // and BF initial size is < 1MB.
+      if (!BF.isSplit() && BF.getSize() < 0x100000) {
```

Comment:
Please avoid magic constants.

Reply:
Moved to separate constant, I think it is OK, isn't it?